authentication/views.py

```diff
@@ -232,3 +240,50 @@ def set_password(self, request, *args, **kwargs):
             update_session_auth_hash(self.request, self.request.user)
             return Response({}, status=status.HTTP_200_OK)
         return response
+
+
+@api_view(["GET", "POST"])
+@permission_classes([IsAuthenticated, IsStaffPermission])
+@authentication_classes([BearerAuthentication])
+def get_user_details_for_keycloak(request, email):
+    """
+    Endpoint for the Keycloak plug-in: https://github.com/daniel-frak/keycloak-user-migration.
+
+    Args:
+        email (string): The email of a user record in Open Discussions.
+
+    Returns:
+        Response: GET requests will respond with the user's first and last name,
+            and email address, along with some additional flags specific for
+            Keycloak.
+            POST requests will respond with a 200 status if the password,
+            contained in the body, matches the user's record.  If the
+            password does not match the user's record, a 403 response is
+            provided.
+            Both requests will respond with a 404 if no Open Discussion's user
+            has an email matching the email argument.
+    """
+    user = User.objects.filter(email=email).all()
+    if user:
+        if request.method == "POST":
+            body = json.loads(request.body)
+            if "password" in body and user[0].check_password(body["password"]):
+                return Response({}, status=status.HTTP_200_OK)
+            else:
+                return Response({}, status=status.HTTP_403_FORBIDDEN)
+        else:
+            response = {
+                "email": user[0].email,
+                "username": user[0].email,
+                "firstName": user[0].first_name,
+                "lastName": user[0].last_name,
+                "enabled": True,
+                "emailVerified": True,
+                "attributes": {},
+                "roles": ["default-roles-olapps"],
+                "groups": [],
+                "requiredActions": [],
+            }
+            return Response(response, status=status.HTTP_200_OK)
+    else:
+        return Response({}, status=status.HTTP_404_NOT_FOUND)
```

Comment:
**Suggestion:** 
```suggestion
        raise Http404("User not found")
```

Seems to be what we do elsewhere, e.g., https://github.com/mitodl/open-discussions/blob/c8d922415e61e707780ddc8000b062b6cd96e475/open_discussions/views.py#L128